app/jobs/event/apply_fee_waiver_job.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class Event
+  class ApplyFeeWaiverJob < ApplicationJob
+    queue_as :low
+
+    DATE_LOCK = Date.new(2025, 11, 1)
+
+    def perform
+      Event.find_each do |event|
+        process_event(event)
+      end
+    end
+
+    private
+
+    def process_event(event)
+      active_teen_count = event.users.active_teenager.count
+
+      if active_teen_count >= 5 && event.fee_waiver_eligible && Date.current < DATE_LOCK
+        plan_type =
+          if active_teen_count >= 10
+            Event::Plan::Standard::FeeWaived
+          else
+            Event::Plan::Standard::ThreePointFive
+          end
+
+        unless event.plan.type == plan_type
```

Comment:
```suggestion
        unless event.plan.instance_of?(plan_type)
```

`event.plan.type` will return a string such as
```ruby
"Event::Plan::FeeWaived"
```
However, `plan_type` will be a class such as
```ruby
Event::Plan::FeeWaived
```

This means `"Event::Plan::FeeWaived" != Event::Plan::FeeWaived`. So, we need to check to compare either class to class or string to string. I suggested using `instance_of?`, which will compare class to class, but any of the following also works:

```ruby
event.plan.type == plan_type.name # compares strings

event.plan.class == plan_type # compares classes, and is the same as instance_of?
```

It's common for `instance_of?` to be confused with `is_a?`. `is_a?(Event::Plan::FeeWaived)` allows for any of `Event::Plan::FeeWaived`'s subclasses, which we do NOT want in this case. We want to check for specifically that class (and none of its subclasses).